platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/componentframework/TaskProcessors.java

```diff
@@ -0,0 +1,211 @@
+     * processor to handle asynchronously.
+     *
+     * @param processorClass
+     * 		the class of the task processor to get the submitter for
+     * @return the submitter for the given task processor class
+     */
+    @SuppressWarnings("unchecked")
+    public <T extends TaskProcessor> @NonNull T getSubmitter(@NonNull final Class<T> processorClass) {
+        Objects.requireNonNull(processorClass);
+        return (T) parts.get(processorClass).getSubmitter();
+    }
+
+    /**
+     * Start the framework. This will start all the task processors in their own threads.
+     */
+    @SuppressWarnings("unchecked")
+    public void start() {
+        throwIfStarted();
+        parts.values().stream().filter(p -> p.getImplementation() == null).forEach(p -> {
+            throw new IllegalStateException(String.format(
+                    "TaskProcessor %s has no implementation set",
+                    p.getDefinition().getName()));
+        });
+
+        for (final ProcessorParts processorParts : parts.values()) {
+            final Map<Class<?>, InterruptableConsumer<?>> processingMethods =
+                    processorParts.getImplementation().getProcessingMethods();
+            final InterruptableConsumer<?> handler;
+            if (processingMethods.size() == 1) {
+                handler = processingMethods.values().iterator().next();
+            } else {
+                handler = new MultiHandler(processingMethods)::handle;
+            }
+
+            final QueueThread<Object> queueThread = defaultConfiguration
+                    .copy()
+                    .setThreadName(processorParts.getConfig().name())
+                    .setQueue(processorParts.getQueue())
+                    .setHandler((InterruptableConsumer<Object>) handler)
+                    .build(true);
+            processorParts.setQueueThread(queueThread);
+        }
+        started = true;
+    }
+
+    /**
+     * Get the queue thread for the given task processor class.
+     *
+     * @param processorClass
+     * 		the class of the task processor to get the queue thread for
+     * @return the queue thread for the given task processor class
+     */
+    public @NonNull QueueThread<?> getQueueThread(@NonNull final Class<? extends TaskProcessor> processorClass) {
+        throwIfNotStarted();
+        return parts.get(processorClass).getQueueThread();
+    }
+
+    /**
+     * Stop the framework. This will stop all the task processors.
+     */
+    public void stop() {
+        throwIfNotStarted();
+        parts.values().forEach(p -> p.getQueueThread().stop());
+    }
+
+    private void throwIfStarted() {
+        if (started) {
+            throw new IllegalStateException("Cannot perform this operation after starting");
+        }
+    }
+
+    private void throwIfNotStarted() {
+        if (!started) {
+            throw new IllegalStateException("Cannot perform this operation before starting");
+        }
+    }
```

Comment:
Once you extend `Lifecycle` there are some convenient default helper methods such as `throwIfNotInPhase(NOT_STARTED)`.